src/common/datetime/calc_date.ts

```diff
@@ -132,15 +132,13 @@ export const shiftDateRange = (
     end = calcDate(endDate, addDays, locale, config, difference);
   } else {
     const difference =
-      ((calcDateDifferenceProperty(
+      (calcDateDifferenceProperty(
         endDate,
         startDate,
         differenceInMilliseconds,
         locale,
         config
-      ) as number) +
-        1) *
-      (forward ? 1 : -1);
+      ) as number) * (forward ? 1 : -1);
```

Comment:
This one makes me a little worried about other unexpected effects, but I cannot justify a good reason to add 1 millisecond to the difference here. 

Generally the end date is considered to be exclusive of the final moment, so I would expect the difference in milliseconds between 4:00:00.000 and 5:00:00.000 to be exactly 1 hour, not 1 hour and 1 millisecond. 

This had implications when using the next/previous date buttons, where if you had a range like 4:00 - 5:00 displayed, and hit previous button, it would change to 2:59 - 3:59, not 3:00 - 4:00 as expected.